php/libraries/NDB_BVL_Instrument.class.inc

```diff
@@ -1574,8 +1587,11 @@ abstract class NDB_BVL_Instrument extends NDB_Page
      */
     function getDataEntryCompletionStatus(): string
     {
-        $data = $this->getInstanceData();
-        return $data["Data_entry_completion_status"];
+        $db = Database::singleton();
+        return $db->pselectOne(
+            "SELECT Data_entry_completion_status FROM flag WHERE CommentID=:CID",
+            ['CID' => $this->getCommentID()]
+        ) ?? '';
```

Comment:
I think this Should default to `Incomplete` or throw an error depending on if this value is always fille din or not